cloud/terraform/machineactuator.go

```diff
@@ -345,7 +345,42 @@ func runTerraformCmd(stdout bool, workingDir string, arg ...string) (bytes.Buffe
 
 func (tf *TerraformClient) Delete(machine *clusterv1.Machine) error {
 	glog.Infof("TERRAFORM DELETE.\n")
-	return nil
+	// Check if the instance exists, return if it doesn't
+	instance, err := tf.instanceIfExists(machine)
+	if err != nil {
+		return err
+	}
+
+	if instance == nil {
+		return nil
+	}
+
+	homedir, err := getHomeDir()
+	if err != nil {
+		return err
+	}
+	machinePath := path.Join(homedir, fmt.Sprintf(".terraform.d/kluster/machines/%s", machine.ObjectMeta.Name))
```

Comment:
nite but we're joining this path in create as well. Can you just create a helper like `getMachinePath`?

Reply:
I would but I expect this to be deleted eventually once we stop using disk for storing state.